File: find-the-duplicate-number/find-the-duplicate-number.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        nums.sort()
        i=0
        j=1
        while i<=len(nums)-2:    
            if nums[i]==nums[j]: 
                return nums[j]
            i+=1
            j+=1
            
            
        
        
        

File: find-the-duplicate-number/find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        unordered_map<int,int>umap; 
        for(int i=0;i<nums.size();i++){
            umap[nums[i]]++; 
        }
        for(auto  i=umap.begin();i!=umap.end();i++){
            if(i->second>1){
                return i->first; 
                
            }
        }
        return 1; 
        
    }
};